File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        n = len(mat)
        
        total = 0
        for rc in range(n):
            total += mat[rc][rc]
            if rc != n - rc - 1:
                total += mat[rc][n - rc - 1]
        return total

File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.java
class Solution {
    public int diagonalSum(int[][] mat) {
        int total = 0;
        for(int i = 0; i < mat.length; i++){
            total += mat[i][i] + mat[i][mat.length - i - 1];
        }
        if((mat.length & 1) == 1)
            return total - mat[mat.length / 2][mat.length / 2];
        else
            return total;
    }
}